docs/development/machine-bootstrap.md

```diff
@@ -0,0 +1,22 @@
+
+## Flow of creating and bootstrapping Machines/Nodes
+The following steps will be ensured by the garden-controller-manager during a creation/reconciliation flow:
+
+- The machine-controller-manager gets deployed into the Shoot namespace of the Seed.
+- The proper ``<cloudprovider>MachineClass`` objects and their corresponding secrets gets deployed in the Shoot namespace of the Seed. Those secrets contain information how to bootstrap the nodes and will be used by the machine-controller-manager to configure cloud-config/user-data of the VM.
+- The ``MachineDeployment``s for the worker pools specified in the Shoot manifest will be applied. Those deployments will result in the required ``Machine`` objects in the Shoot namespace of the Seed.
+- A secret, which contain a valid bootstrap token and
+the cloud-config configuration for bootstrapping nodes, will be created in the kube-system namespace of the Shoot.
+
+The machine-controller-manager will pick up the ``Machine`` objects and create the VMs on the respective cloud provider. Bootstrapping VMs is done via OS-specific configuration (e.g., [CoreOS cloud-config]((https://coreos.com/os/docs/latest/cloud-config.html))).
+
+The cloud-config will be downloaded from the Shoot API server. This is achieved by copying the contents of the machineclass secret onto the machine. This secret contains configuration to register a systemd service ``cloud-config-downloader`` and a kubeconfig to fetch the secret, which contain the cloud-config and the bootstrap-token, from the Shoot API server. The downloaded cloud-config configuration does contain the actual cloud-config, which contains all the services and configurations for the VM to register as Kubernetes node.
+
+The kubelet requires a suitable kubeconfig to join the cluster and it will generate it by relying on the [tls-bootstraping](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping/) functionality of the kubelet.
+
+#### Details
+The ``cloud-config-downloader`` fetches the bootstap-token from Shoot API server and checks, if already a kubeconfig file exists on the path, which the kubelet configures via its ``--kubeconfig`` flag. If the file exists the downloader does nothing, because it assumes that a proper kubeconfig already exists. If the file not exists, then a bootstrap-kubeconfig will be created with the bootstrap token to authenticate. This bootstrap-kubeconfig will be placed on the path configured via kubelets [``--bootstrap-kubeconfig``](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/) flag. Afterwards, the kubelet will perform the tls-bootstrapping process to generate a suitable kubeconfig, which will be stored on the path, which kubelets ``--kubeconfig`` flag is pointing to.
```

Comment:
s/bootstap-token/bootstrap-token

Reply:
no comma before "if already a kubeconfig..." and no comma after "on the path" and no comma after "does nothing"

Reply:
s/not exists,/does not exist